### PR TITLE
Add type hints for collection expressions

### PR DESCRIPTION
**Pull request by akhera99**
Fixes: #72250 

image

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Awesome!  Can you show a picture of the options?

File: src/Features/CSharp/Portable/InlineHints/CSharpInlineTypeHintsService.cs [hunk 1]
```diff
@@ -101,6 +102,19 @@ public CSharpInlineTypeHintsService()
...
 
+        if (forCollectionExpressions || displayAllOverride)
+        {
+            if (node is CollectionExpressionSyntax collectionExpression)
+            {
+                var type = semanticModel.GetTypeInfo(collectionExpression, cancellationToken).ConvertedType;
+                if (IsValidType(type))
+                {
+                    var span = new TextSpan(collectionExpression.OpenBracketToken.SpanStart, 0);
+                    return new(type, span, new TextChange(span, " " + type.ToDisplayString(s_minimalTypeStyle)), leadingSpace: true);
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/InlineHints/CSharpInlineTypeHintsService.cs:113]: Why the extra space?

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/CSharp/Portable/InlineHints/CSharpInlineTypeHintsService.cs (hunk 1 above)
Comment (akhera99) [reply src/Features/CSharp/Portable/InlineHints/CSharpInlineTypeHintsService.cs:113]: you're right, unnecessary

**Comment by akhera99** [comment]
> Awesome! Can you show a picture of the options?

image

**Pull request merged**